packages/khalti/lib/src/khalti_http_client.dart

```diff
@@ -6,16 +6,32 @@ import 'package:http/http.dart' as http;
 import 'package:khalti/khalti.dart';
 import 'package:khalti_core/khalti_core.dart';
 
+/// Typedef for http.Client that is used internally.
+typedef InnerClient = http.Client;
+
 /// The default [KhaltiClient] implementation.
 ///
 /// Uses [http](https://pub.dev/packages/http) package under-the-hood.
 class KhaltiHttpClient extends KhaltiClient {
+  final InnerClient? _client;
```

Comment:
This can be non-null. Also please move the property below constructor.

Reply:
I thought not having to pass `innerClient` explicitly would be nice as it would eventually be instantiated anyway. What do you think?

Reply:
Making this non-null won't force us to send the inner client.

Reply:
Oh. I thought you were talking about the constructor parameter. Resolved.